R/teal_slice.R

```diff
@@ -336,25 +336,61 @@ is.teal_slices <- function(x) { # nolint
 }
 
 
-# JSON utils ------------------------------------------------------------------------------------------------------
+# utils -----------------------------------------------------------------------------------------------------------
 
-slice_to_json <- function(x) {
+trim_character <- function(x, nchar) {
+  json_t <- substr(x, 1, nchar)
+  substr(json_t, nchar - 2, 40) <- "..."
+  json_t
+}
+
+# JSON utils for teal_slice ---------------------------------------------------------------------------------------
+
+slice_list_to_json <- function(x) {
+  # selected and choices should remain arrays
   if (!is.null(x$selected)) x$selected <- I(x$selected)
   if (!is.null(x$choices)) x$choices <- I(x$choices)
 
   jsonlite::toJSON(x, pretty = TRUE, auto_unbox = TRUE, digits = 16, null = "null")
 }
 
-trim_character <- function(x, nchar) {
-  json_t <- substr(x, 1, nchar)
-  substr(json_t, nchar - 2, 40) <- "..."
-  json_t
+center_slice_json_split <- function(json) {
+  format_name <- function(n) {
+    if (nchar(n) == 1) {
+      return(n)
+    } else {
+      paste(format(n, width = name_width), ":")
+    }
+  }
+
+  json_s <- strsplit(json, split = ":", fixed = TRUE)
+  name_width <- max(unlist(gregexpr(":", json))) - 1
+
+  vapply(json_s, function(x) paste0(format_name(x[1]), na.omit(x[2])), character(1))
+}
+
+
+
+# JSON utils for teal_slices --------------------------------------------------------------------------------------
+
+slices_list_to_json <- function(tss) {
```

Comment:
Ah, I saw `tss` as the argument and assumed `teal_slices` because that is what it it in `slices_field`. If it is just a list, then the function name is fine :+1:  but please change the argument to `x`.

Reply:
got it 3926e6570e8fc7f8d991da2d24eeacb8ed44fcd5